site/guides/tenant-onboarding.md

```diff
@@ -0,0 +1,274 @@
+
+```shell
+
+pkg-catalog $ kpt pkg tree tenant
+Package "tenant"
+├── [Kptfile]  Kptfile tenant
+├── [namespace.yaml]  Namespace tenant-name
+├── [ns-invariant.yaml]  StarlarkRun ns-invariant
+├── [quota.yaml]  ResourceQuota tenant-name/quota
+├── [role-binding.yaml]  RoleBinding tenant-name/sa-admin
+└── [service-account.yaml]  ServiceAccount tenant-name/sa
+
+```
+
+The tenant package’s Kptfile offers basic customization and enforces single
+namespace constraint.
+
+```shell
+pkg-catalog $ cat tenant/Kptfile
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: tenant
+info:
+  description: Base tenant package
+pipeline:
+  mutators:
+    - image: set-namespace:v0.1
+      configMap:
+        namespace: tenant-name # ←- will be customized for pkg variant
+  validators:
+    - image: gcr.io/kpt-fn/starlark:v0.3
+      configPath: ns-invariant.yaml
+
+```
+
+### Optional resources
+
+Sometimes customizing a package may require adding a new resource. For example,
+if a team wants to define rules for isolating the network for a tenant, they
+need to define network policy resources. Our tenant package has a hidden
+directory called `.snippets` that contains snippets of such resources that a
+package consumer may need. Platform team can also surface canned configuration
+for example quota.medium.yaml can be used for medium size workloads.
+
+Also `.krmignore` files instructs kpt to ignore all resources in the `.snippets`
+directory for other workflows.
+
+```shell
+pkg-catalog $ ls tenant/.snippets/
+isolate-network-policy.yaml  quota.large.yaml  quota.medium.yaml
+quota.small.yaml  role-binding.yaml  service-account.yaml
+
+pkg-catalog $ cat tenant/.krmignore
+.snippets
+
+```
+
+### Publishing Tenant Package
+
+So once you are happy with the tenant package, you can publish the tenant
+package by tagging the version as shown below:
+
+```shell
+# Assuming you are in the pkg catalog repo where tenant package exists
+
+# Pl. remove the inventory information from the Kptfile before publishing
+
+# create new tag
+$ git tag tenant/v0.1 main
+
+# push the tag to the upstream
+$ git push origin tenant/v0.1
+
+```
+
+## Tenant onboarding workflow
+
+Now, let’s take a look at how tenant onboarding will work. Steps described below
+can be done by a member of the platform team or application developer team. The
+good thing is that it enables self-service workflow for application teams where
+they can request for new tenants by simply issuing a PR against the platform repo.
+
+```shell
+
+# assuming you have already a fork of the platform repo
+# (remote origin pointing to the fork).
+
+# create a new branch to onboard new tenant
+$ git checkout -b onboarding-tenant-a
+
+$ cd tenants
+$ kpt pkg get <pkg-catalog-repo>/tenant@v0.1 tenant-a
+
+# tenant customizations:
+# change the namespace to tenant-a in the tenant-a/Kptfile
+# configure the quota.yaml if needed or
+# or cp .snippets/quota.{small, large, medium}.yaml as quota.yaml
```

Comment:
if these are just changes in numbers possibly also we don't need a snipper for it but rather show how to do this with search and replace function?

Reply:
Most of the time, developer teams look for guidance from platform teams for different customizations. This is one way to package canned configuration that developers team can map to their requirement. So, in a way, it is solving "what numbers should I fill in my quota.yaml" problem. If they know the numbers, then yes, direct editing or using `search-replace` is always an option.